lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+        results['portgroup'] = self.portgroup_name
+        if self.module.check_mode:
+            results['result'] = "Portgroup would be deleted"
+        else:
+            try:
+                task = self.dvs_portgroup.Destroy_Task()
+            except vim.fault.VimFault as vim_fault:
+                self.module.fail_json(msg="Failed to deleted portgroup : %s" % to_native(vim_fault))
+            wait_for_task(task)
+            results['result'] = "Portgroup deleted"
+        self.module.exit_json(**results)
 
-        if not self.module.check_mode:
-            changed, result = self.create_port_group()
-        self.module.exit_json(changed=changed, result=str(result))
+    def exit_unchanged(self):
+        """Exit with status message"""
+        changed = False
+        results = dict(changed=changed)
+        results['dvswitch'] = self.switch_name
+        results['portgroup'] = self.portgroup_name
+        results['result'] = "Portgroup not present"
+        self.module.exit_json(**results)
 
-    def check_dvspg_state(self):
-        self.dv_switch = find_dvs_by_name(self.content, self.module.params['switch_name'])
+    def update_port_group(self):
+        """Check and update DVS portgroup"""
+        changed = changed_policy = changed_vlan = changed_vlan_trunk_range = \
+            changed_security = changed_teaming = changed_failover_order = False
+        results = dict(changed=changed)
+        results['dvswitch'] = self.switch_name
+        results['portgroup'] = self.portgroup_name
+        changed_list = []
 
-        if self.dv_switch is None:
-            self.module.fail_json(msg="A distributed virtual switch with name %s does not exist" % self.module.params['switch_name'])
-        self.dvs_portgroup = find_dvspg_by_name(self.dv_switch, self.module.params['portgroup_name'])
+        pg_spec = vim.dvs.DistributedVirtualPortgroup.ConfigSpec()
+        # Use the same version in the new spec; The version will be increased by one by the API automatically
+        pg_spec.configVersion = self.dvs_portgroup.config.configVersion
+        pg_config = self.dvs_portgroup.config
 
-        if self.dvs_portgroup is None:
-            return 'absent'
+        # Check number of ports (only if option is specified)
+        # NOTE: 'self.portgroup_type' is deprecated. Will be removed in 2.11
+        if self.port_binding == 'ephemeral' or self.portgroup_type == 'ephemeral':
+            results['num_ports'] = 'n/a'
         else:
-            return 'present'
+            results['num_ports'] = self.num_ports
+            if self.num_ports and pg_config.numPorts != self.num_ports:
+                changed = True
+                changed_list.append("num ports")
+                results['num_ports_previous'] = pg_config.description
+                pg_spec.numPorts = self.num_ports
+
+        # Check port binding
+        # NOTE: 'self.portgroup_type' is deprecated. Will be removed in 2.11
+        if self.portgroup_type:
```

Comment:
Same comment as above, if both `portgroup_type` and `port_binding` are set, it will ignore one... it should check for this and fail if there is a conflict

Reply:
Like described above, the module shouldn't fail if a deprecated parameter will be used. To be honest, I didn't check if the play with the deprecated options will work, but at least the intention is to not fail with an error.